homeassistant/components/onvif/config_flow.py

```diff
@@ -0,0 +1,311 @@
+                    configured = True
+                    break
+            if not configured:
+                self.devices.append(device)
+
+        LOGGER.debug("Discovered ONVIF devices %s", pformat(self.devices))
+
+        if self.devices:
+            names = []
+
+            for device in self.devices:
+                names.append(f"{device[CONF_NAME]} ({device[CONF_HOST]})")
+
+            names.append(CONF_MANUAL_INPUT)
+
+            return self.async_show_form(
+                step_id="device",
+                data_schema=vol.Schema({vol.Optional(CONF_HOST): vol.In(names)}),
+            )
+
+        return await self.async_step_manual_input()
+
+    async def async_step_manual_input(self, user_input=None):
+        """Manual configuration."""
+        if user_input:
+            self.onvif_config = user_input
+            return await self.async_step_auth()
+
+        return self.async_show_form(
+            step_id="manual_input",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_NAME): str,
+                    vol.Required(CONF_HOST): str,
+                    vol.Required(CONF_PORT, default=DEFAULT_PORT): int,
+                }
+            ),
+        )
+
+    async def async_step_auth(self, user_input=None):
+        """Username and Password configuration for ONVIF device."""
+        if user_input:
+            self.onvif_config[CONF_USERNAME] = user_input[CONF_USERNAME]
+            self.onvif_config[CONF_PASSWORD] = user_input[CONF_PASSWORD]
+            return await self.async_step_profiles()
+
+        return self.async_show_form(
+            step_id="auth",
+            data_schema=vol.Schema(
+                {vol.Required(CONF_USERNAME): str, vol.Required(CONF_PASSWORD): str}
+            ),
+        )
+
+    async def async_step_profiles(self, user_input=None):
+        """Fetch ONVIF device profiles."""
+        errors = {}
+
+        LOGGER.debug(
+            "Fetching profiles from ONVIF device %s", pformat(self.onvif_config)
+        )
+
+        device = get_device(
+            self.hass,
+            self.onvif_config[CONF_HOST],
+            self.onvif_config[CONF_PORT],
+            self.onvif_config[CONF_USERNAME],
+            self.onvif_config[CONF_PASSWORD],
+        )
+
+        await device.update_xaddrs()
+
+        try:
+            # Get the MAC address to use as the unique ID for the config flow
+            if not self.device_id:
+                devicemgmt = device.create_devicemgmt_service()
+                network_interfaces = await devicemgmt.GetNetworkInterfaces()
+                for interface in network_interfaces:
+                    if interface.Enabled:
+                        self.device_id = interface.Info.HwAddress
+
+            if self.device_id is None:
+                return self.async_abort(reason="no_mac")
+
+            await self.async_set_unique_id(self.device_id)
```

Comment:
If the user initiates this PR, we should not raise if any other are in progress. Default is true. 

```suggestion
            await self.async_set_unique_id(self.device_id, raise_on_progress=False)
```